### PR TITLE
Rename `fields-discriminated` to `fields-to-discriminateBy`

### PR DESCRIPTION
**Pull request by apelisse**

### CODE REVIEW

**Comment by sttts** [comment]
I prefer `fields` very much.

/approve
lgtm

Giving @lavalamp a chance for a final look.

File: test/integration/testdata/golden.json [hunk 1]
```diff
@@ -402,7 +402,7 @@
     "x-kubernetes-unions": [
      {
       "discriminator": "unionType",
-      "fields-discriminated": {
+      "fields": {
```
Comment (lavalamp) [inline test/integration/testdata/golden.json]: I think that just plain "fields" is going to be confusing, which is first, the field or its discriminator? How do I even know the second value is a discriminator?

How about "fields-to-discriminator-value"? It's really wordy :/ but I think just "fields" doesn't help people enough.

field-names? fields-to-names?

Comment (apelisse) [reply test/integration/testdata/golden.json]: Out of all these options, I like `fields-discriminated` most :-)

Comment (apelisse) [reply test/integration/testdata/golden.json]: Let's use `fields-to-discriminatedBy`?

Comment (apelisse) [reply test/integration/testdata/golden.json]: Renamed, PTAL

Comment (apelisse) [reply test/integration/testdata/golden.json]: renamed to `fields-to-discriminateBy`

**Pull request merged**